include/velocypack/Iterator.h

```diff
@@ -87,7 +87,7 @@ class ArrayIterator {
 #if (defined(_MSC_VER) ||                                     \
      (defined(__clang_version__) && __clang_major__ >= 12) || \
      (!defined(__clang__) && defined(__GNUC__) && __GNUC__ >= 10))
-  [[nodiscard]] std::default_sentinel_t end() const noexcept { return {}; }
+  [[nodiscard]] static constexpr std::default_sentinel_t end() { return {}; }
```

Comment:
why noexcept was removed?

Btw not sure is a good idea.
Some checkers like clang tidy sometimes make warnings about calling static function like
object.func instead of Object::func

For example check this https://github.com/llvm/llvm-project/blob/main/libcxx/include/__coroutine/trivial_awaitables.h